modules/core/src/lib/draw-layers.js

```diff
@@ -247,25 +252,60 @@ function drawLayersInViewport(
   logRenderStats({renderStats, pass, redrawReason, stats});
 }
 
+function getLightSources({viewport, effects}) {
+  let lightEffect;
+  if (effects && Array.isArray(effects)) {
+    lightEffect = effects.find(effect => effect instanceof LightingEffect);
+  }
+
+  const lightSources = {
+    ambientLight: null,
+    pointLights: [],
+    directionalLights: []
+  };
+
+  if (lightEffect && lightEffect.ambientLight) {
+    lightSources.ambientLight = lightEffect.ambientLight;
+  }
+
+  if (lightEffect && lightEffect.pointLights) {
+    lightSources.pointLights = lightEffect.getProjectedPointLights(viewport);
+  }
+
+  if (lightEffect && lightEffect.directionalLights) {
+    lightSources.directionalLights = lightEffect.directionalLights;
+  }
+  return lightSources;
+}
+
 function drawLayerInViewport({
   gl,
   layer,
   layerIndex,
   drawPickingColors,
   pixelRatio,
   glViewport,
-  parameters
+  parameters,
+  effects
 }) {
-  const moduleParameters = Object.assign(Object.create(layer.props), {
-    viewport: layer.context.viewport,
-    pickingActive: drawPickingColors ? 1 : 0,
-    devicePixelRatio: pixelRatio
-  });
+  const moduleParameters = Object.assign(
+    Object.create(layer.props),
+    {
+      viewport: layer.context.viewport,
+      pickingActive: drawPickingColors ? 1 : 0,
+      devicePixelRatio: pixelRatio,
+      effects
+    },
+    getLightSources({
+      viewport: layer.context.viewport,
```

Comment:
Interesting... perhaps we may want to be able to specify different effects per viewport? Something to think about as we build out the effect manager.